sdk/core/Azure.Core.TestFramework/src/RecordedTestBase.cs

```diff
@@ -154,5 +166,65 @@ public virtual void StopTestRecording()
 #endif
             Recording?.Dispose(save);
         }
+
+        protected ValueTask<Response<T>> WaitForCompletionAsync<T>(Operation<T> operation)
+        {
+            if (Mode == RecordedTestMode.Playback)
+            {
+                return operation.WaitForCompletionAsync(ZeroPollingInterval, default);
+            }
+            else
+            {
+                return operation.WaitForCompletionAsync();
+            }
+        }
+
+        protected ResourcesManagementClient GetResourceManagementClient()
+        {
+            var options = InstrumentClientOptions(new ResourcesManagementClientOptions());
+            CleanupPolicy = new ResourceGroupCleanupPolicy();
+            options.AddPolicy(CleanupPolicy, HttpPipelinePosition.PerCall);
+
+            return CreateClient<ResourcesManagementClient>(
+                TestEnvironment.SubscriptionId,
+                TestEnvironment.Credential,
+                options);
+        }
+
+        protected async Task CleanupResourceGroupsAsync()
+        {
+            if (CleanupPolicy != null && Mode != RecordedTestMode.Playback)
+            {
+                var resourceGroupsClient = new ResourcesManagementClient(
+                    TestEnvironment.SubscriptionId,
+                    TestEnvironment.Credential,
+                    new ResourcesManagementClientOptions()).ResourceGroups;
+                foreach (var resourceGroup in CleanupPolicy.ResourceGroupsCreated)
+                {
+                    await resourceGroupsClient.StartDeleteAsync(resourceGroup);
+                }
+            }
+        }
+
+        protected async Task<string> GetFirstUsableLocationAsync(ProvidersOperations providersClient, string resourceProviderNamespace, string resourceType)
+        {
+            var provider = (await providersClient.GetAsync(resourceProviderNamespace)).Value;
+            return provider.ResourceTypes.Where(
+                (resType) =>
+                {
+                    if (resType.ResourceType == resourceType)
+                        return true;
+                    else
+                        return false;
+                }
+                ).First().Locations.FirstOrDefault();
+        }
+
+        protected void SleepInTest(int milliSeconds)
```

Comment:
What cases do we want to sleep? In general we try to avoid sleeping but instead usually do some sort of polling.

Reply:
Again just moved it over for now. Can change in separate PR that will have the larger changes discussed. This is just to reduce the size of that future PR.

Reply:
We are staying with parity with this PR

Reply:
I recommend a tracking bug you can link from these comments. It's a fairly consistent way we use to keep track of improvements to make from PRs.

Reply:
Good idea @heaths 

I will make issues for these when we make further changes in the future.